legofy/__init__.py

```diff
@@ -96,46 +106,82 @@ def get_new_size(base_image, brick_image, bricks=None):
 
     return new_size
 
+def legofy_gif_pythononly(base_image, brick_image, output_path, bricks):
+    '''Alternative function that legofies animated gifs, makes use of images2gif - uses numpy!'''
+    im = base_image
+
+    # Read original image duration
+    original_duration = im.info['duration']
+
+    # Split image into single frames
+    frames = [frame.copy() for frame in ImageSequence.Iterator(im)]
+
+    # Create container for converted images
+    frames_converted = []
+
+    print("Number of frames to convert: " + str(len(frames)))
+
+    i = 0
+    # Iterate through single frames
+    for frame in frames:
+        i += 1
+
+        print("Converting frame number " + str(i))
+
+        new_size = get_new_size(frame, brick_image, bricks)
+        frame = frame.convert("RGB")
+        if new_size != frame.size:
+            frame.thumbnail(new_size, Image.ANTIALIAS)
+        new_frame = make_lego_image(frame, brick_image)
+        frames_converted.append(new_frame)
+
+    # Make use of images to gif function - slightly modified python 3.0 port of
+    images2gif.writeGif(output_path, frames_converted, duration=original_duration/1000.0, dither=0)
 
 def legofy_gif(base_image, brick_image, output_path, bricks):
     '''Legofy an animated GIF'''
-    new_size = get_new_size(base_image, brick_image, bricks)
-    tmp_dir = os.path.join(os.path.dirname(__file__), "tmp_frames")
-    # Clean up tmp dir if it exists
-    if os.path.exists(tmp_dir):
+
+    if os.name == "nt" and os.environ.get('MAGICK_HOME') == None:
+        print('Could not find the MAGICK_HOME environment variable. Trying to use legofy_gif_pythononly().')
+        legofy_gif_pythononly(base_image, brick_image, output_path, bricks)
```

Comment:
Well it works. 

I'm too a fan of platform independent solutions but I wanted to leave the final decision to JuanPotato. Perhaps you are right.

Btw. thanks for the code review.